base/src/main/java/org/mozilla/jss/asn1/BIT_STRING.java

```diff
@@ -115,14 +115,14 @@ public byte[] getBits() {
      * will not accurately reflect the number of bits in the BIT STRING,
      * because the size of a BitSet is always rounded up to the next multiple
      * of 64. The extra bits will be set to 0.
-     * 
+     *
      * @return BIT STRING as BitSet.
      */
     public BitSet toBitSet() {
         BitSet bs = new BitSet();
         int numBits = (bits.length * 8) - padCount;
         for (int i = 0; i < numBits; i++) {
-            if ((bits[i / 8] & (0x80 >>> (i % 8))) != 0) {
+            if (((bits[i / 8] | 0xff) & (0x80 >>> (i % 8))) != 0) {
```

Comment:
Hmm.. shouldn't we use `& 0xff` instead of `| 0xff` to mask off the bits? Also, IIUC the code is already masking off the bits using `& (0x80 ...)` so the additional mask might not be necessary.

Reply:
@edewata yes, you are right about the mask, my mistake. It has to be with `&`.

I think the code analysis highlight the problem when you do bitwise operations between bytes and integers. In this particular case we have an additional mask so the promotion should not be a problem. If you and @ckelleyRH  think it is better to maintain the code we can disable this bug on Sonarcloud. Alternatively, I could try to modify the code in order to make the second mask identifiable by the code analysis but I am not sure about it.

Reply:
Is it possible to disable SonarCloud for a specific line like `# pylint: disable=...`? I'm not sure it's worth tweaking the code just to prevent false positives in SonarCloud.

Reply:
> Is it possible to disable SonarCloud for a specific line

[Yes](https://github.com/dogtagpki/tomcatjss/blob/master/core/src/main/java/org/apache/tomcat/util/net/jss/TomcatJSS.java#L519)! Just add `//NOSONAR` to a line to disable it, and I agree we should.

Reply:
OK. I have reverted the code and disabled the checks in the lines.

The web interface allows to set a bug as a false positive but it is not clear what happen when the code get  updated and I did not find documentation.